### PR TITLE
A4A: Check if agency website is active during signup

### PR DESCRIPTION
**Pull request by travisw**
Related to: https://github.com/Automattic/automattic-for-agencies-dev/issues/1589
PT: pfunGA-3ie-p2

## Proposed Changes

* When submitting the signup form make an API request to check if the agency website is actually an active site.

## Why are these changes being made?

* This is part of the changes we're introducing to mitigate fraud by introducing some intentional friction in the process. 

## Testing Instructions

* Test this with the backend changes in D168665-code and while sandboxing `public-api.wordpress.com`
* As a logged-out user, visit /signup
* Fill in all required fields
* For the "Business URL" field test 4 different things (keep an eye on network requests in the browser console):
  1. An invalid URL (eg. `travisblog`) 
  This should act as it did before, showing the error "Please enter a valid URL" and not making any new network requests.
  Screenshot 2024-12-13 at 1 22 11 PM

  2. A valid URL to facebook.com, linkedin.com, or instagram.com (eg. `facebook.com/user`)
  This should show the error "Please enter a valid site URL for your business. If you're experiencing issues contact us at partnerships@automattic.com" and not making any new network requests.
  Screenshot 2024-12-17 at 10 54 46 AM

  3. A valid URL that **isn't** a real active site (eg. `travisinvalid.blog`)
  This should make a request to the new endpoint and show the error "Please enter a valid site URL for your business. If you're experiencing issues contact us at partnerships@automattic.com"
  Screenshot 2024-12-17 at 10 54 29 AM

  4. A valid URL that **is** a real active site (eg. `travis.blog`)
  This should make a request to the new endpoint, not show any validation error, and continue to the next signup step.
    * Also test this as a logged-in user without an existing A4A account. You should similarly proceed to the next signup step.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17062559

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @travisw for including a screenshot in the description! This is really helpful for our translators.

File: client/a8c-for-agencies/sections/signup/agency-details-form/hooks/use-signup-form-validation.ts [hunk 1]
```diff
@@ -1,9 +1,12 @@
 import { useTranslate } from 'i18n-calypso';
 import { useCallback, useState } from 'react';
+import { isSiteActive } from 'calypso/a8c-for-agencies/components/form/utils';
 import { AgencyDetailsPayload } from '../types';
 
 export const CAPTURE_URL_RGX =
 	/^(http:\/\/www\.|https:\/\/www\.|http:\/\/|https:\/\/)?[a-z0-9]+([-.][a-z0-9]+)*\.[a-z]{2,63}(:[0-9]{1,5})?(\/.*)?$/i;
+const CAPTURE_SOCIAL_URL_RGX =
```
Comment (andrii-lysenko) [inline client/a8c-for-agencies/sections/signup/agency-details-form/hooks/use-signup-form-validation.ts:9]: Nice.

**Review by andrii-lysenko — APPROVED** [review]
Looks and works great!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17062559).

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17062559).

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17062559).

**Pull request merged**